File: projet/src/main/java/com/enicar/projet/dtos/ConventionRequestDTO.java
package com.enicar.projet.dtos;


import lombok.Data;

@Data
public class ConventionRequestDTO {

    // ── Entreprise ────────────────────────────────────────────────────
    private String entreprise;

    private String adresseEntreprise;

    private String representantEntreprise;

    private String tuteurStage;

    private String emailEntreprise;

    private String telephoneEntreprise;

    private String faxEntreprise;           // optionnel

    // ── Étudiant ──────────────────────────────────────────────────────

    private String nomEtudiant;


    private String prenomEtudiant;

    private String cin;

    private String telephone;              // téléphone étudiant

    private String email;                  // optionnel pour le PDF


    private String specialite;             // filière affichée dans la convention

    // ── Stage ─────────────────────────────────────────────────────────
    private String dateDebut;              // format dd/MM/yyyy

    private String dateFin;               // format dd/MM/yyyy

    // ── Cases à cocher (type de formation) ────────────────────────────
    private boolean ing;
    private boolean mastere;

    // ── Cases à cocher (spécialité) ───────────────────────────────────
    private boolean info;
    private boolean electrique;
    private boolean indus;

    // ── Cases à cocher (année) ────────────────────────────────────────
    private boolean premiere;
    private boolean deuxieme;
}

File: front_spring/src/app/models/ConventionRequestDTO.ts
// src/app/models/convention-request.model.ts

export interface ConventionRequestDTO {
  // Entreprise
  entreprise: string;
  adresseEntreprise: string;
  representantEntreprise: string;
  tuteurStage: string;
  emailEntreprise: string;
  telephoneEntreprise: string;
  faxEntreprise?: string; // optionnel

  // Étudiant
  nomEtudiant: string;
  prenomEtudiant: string;
  cin: string;
  telephone: string;
  email?: string; // optionnel pour le PDF
  specialite: string;

  // Stage
  dateDebut: string; // format dd/MM/yyyy
  dateFin: string;   // format dd/MM/yyyy

  // Type de formation
  ing: boolean;
  mastere: boolean;

  // Spécialité
  info: boolean;
  electrique: boolean;
  indus: boolean;

  // Année
  premiere: boolean;
  deuxieme: boolean;
}